File: D/520-detect-capital/main.go
package main

import "fmt"

func detectCapitalUse(word string) bool {
	isAllUpper := true
	for i := 0; i < len(word); i++ {
		if word[i] >= 'a' && word[i] <= 'z' {
			isAllUpper = false
			break
		}
	}

	isAllLower := true
	for i := 0; i < len(word); i++ {
		if word[i] >= 'A' && word[i] <= 'Z' {
			isAllLower = false
			break
		}
	}

	isAllInitial := true
	if word[0] >= 'A' && word[0] <= 'Z' {
		for i := 1; i < len(word); i++ {
			if word[i] >= 'A' && word[i] <= 'Z' {
				isAllInitial = false
				break
			}
		}
	} else {
		isAllInitial = false
	}

	return isAllUpper || isAllLower || isAllInitial
}

func main() {
	fmt.Println("a")
}

File: D/520-detect-capital/main.rs
struct Solution {}

impl Solution {
  pub fn detect_capital_use(word: String) -> bool {
    let bb = word.as_bytes().to_vec();
    let mut cnt: i32 = 0;
    let mut cap: bool = false;
    bb.iter().enumerate().for_each(|(idx, &b)| {
      if b >= b'A' && b <= b'Z' {
        cnt += 1;
      }
      if idx == 0 && b >= b'A' && b <= b'Z' {
        cap = true;
      }
    });
    cnt == bb.len() as i32 || cnt == 0 || bb.len() > 1 && cnt == 1 && cap
  }
}
